In [1]:
import pandas as pd

df = pd.read_csv("churn-bigml-20.csv")
df.head()

,State,Account length,Area code,International plan,Voice mail plan,Number vmail messages,Total day minutes,Total day calls,Total day charge,Total eve minutes,Total eve calls,Total eve charge,Total night minutes,Total night calls,Total night charge,Total intl minutes,Total intl calls,Total intl charge,Customer service calls,Churn
0,LA,117,408,No,No,0,184.5,97,31.37,351.6,80,29.89,215.8,90,9.71,8.7,4,2.35,1,False
1,IN,65,415,No,No,0,129.1,137,21.95,228.5,83,19.42,208.8,111,9.40,12.7,6,3.43,4,True
2,NY,161,415,No,No,0,332.9,67,56.59,317.8,97,27.01,160.6,128,7.23,5.4,9,1.46,4,True
3,SC,111,415,No,No,0,110.4,103,18.77,137.3,102,11.67,189.6,105,8.53,7.7,6,2.08,2,False
4,HI,49,510,No,No,0,119.3,117,20.28,215.1,109,18.28,178.7,90,8.04,11.1,1,3.00,1,False


In [2]:
df.shape

(667, 20)

In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 667 entries, 0 to 666
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   State                   667 non-null    str    
 1   Account length          667 non-null    int64  
 2   Area code               667 non-null    int64  
 3   International plan      667 non-null    str    
 4   Voice mail plan         667 non-null    str    
 5   Number vmail messages   667 non-null    int64  
 6   Total day minutes       667 non-null    float64
 7   Total day calls         667 non-null    int64  
 8   Total day charge        667 non-null    float64
 9   Total eve minutes       667 non-null    float64
 10  Total eve calls         667 non-null    int64  
 11  Total eve charge        667 non-null    float64
 12  Total night minutes     667 non-null    float64
 13  Total night calls       667 non-null    int64  
 14  Total night charge      667 non-null    float64
 15  

In [4]:
df.isnull().sum()

State                     0
Account length            0
Area code                 0
International plan        0
Voice mail plan           0
Number vmail messages     0
Total day minutes         0
Total day calls           0
Total day charge          0
Total eve minutes         0
Total eve calls           0
Total eve charge          0
Total night minutes       0
Total night calls         0
Total night charge        0
Total intl minutes        0
Total intl calls          0
Total intl charge         0
Customer service calls    0
Churn                     0
dtype: int64

In [5]:
df.duplicated().sum()

np.int64(0)

In [6]:
df.duplicated().sum()

np.int64(0)

In [7]:
# Work on a copy so the original stays untouched
clean = df.copy()

# Remove duplicate rows (none were found, kept as a safeguard)
clean = clean.drop_duplicates()

# Tidy column names: lowercase, underscores instead of spaces
clean.columns = clean.columns.str.strip().str.lower().str.replace(" ", "_")

# Convert Yes/No text to 1/0
clean["international_plan"] = clean["international_plan"].map({"Yes": 1, "No": 0})
clean["voice_mail_plan"] = clean["voice_mail_plan"].map({"Yes": 1, "No": 0})

# Convert True/False to 1/0
clean["churn"] = clean["churn"].astype(int)

# Make sure State is consistent text
clean["state"] = clean["state"].str.strip().str.upper()

clean.head()

,state,account_length,area_code,international_plan,voice_mail_plan,number_vmail_messages,total_day_minutes,total_day_calls,total_day_charge,total_eve_minutes,total_eve_calls,total_eve_charge,total_night_minutes,total_night_calls,total_night_charge,total_intl_minutes,total_intl_calls,total_intl_charge,customer_service_calls,churn
0,LA,117,408,0,0,0,184.5,97,31.37,351.6,80,29.89,215.8,90,9.71,8.7,4,2.35,1,0
1,IN,65,415,0,0,0,129.1,137,21.95,228.5,83,19.42,208.8,111,9.40,12.7,6,3.43,4,1
2,NY,161,415,0,0,0,332.9,67,56.59,317.8,97,27.01,160.6,128,7.23,5.4,9,1.46,4,1
3,SC,111,415,0,0,0,110.4,103,18.77,137.3,102,11.67,189.6,105,8.53,7.7,6,2.08,2,0
4,HI,49,510,0,0,0,119.3,117,20.28,215.1,109,18.28,178.7,90,8.04,11.1,1,3.00,1,0


In [8]:
clean.info()
print("Rows before:", len(df), "| Rows after:", len(clean))

<class 'pandas.DataFrame'>
RangeIndex: 667 entries, 0 to 666
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   state                   667 non-null    str    
 1   account_length          667 non-null    int64  
 2   area_code               667 non-null    int64  
 3   international_plan      667 non-null    int64  
 4   voice_mail_plan         667 non-null    int64  
 5   number_vmail_messages   667 non-null    int64  
 6   total_day_minutes       667 non-null    float64
 7   total_day_calls         667 non-null    int64  
 8   total_day_charge        667 non-null    float64
 9   total_eve_minutes       667 non-null    float64
 10  total_eve_calls         667 non-null    int64  
 11  total_eve_charge        667 non-null    float64
 12  total_night_minutes     667 non-null    float64
 13  total_night_calls       667 non-null    int64  
 14  total_night_charge      667 non-null    float64
 15  

In [9]:
clean.to_csv("cleaned_churn.csv", index=False)

Summary: The dataset has 667 rows and 20 columns. No missing values and no duplicate rows were found. I standardized column names, converted International plan, Voice mail plan and Churn into 1/0, and saved the result as cleaned_churn.csv

In [10]:
clean[["international_plan", "voice_mail_plan", "churn"]].head()

,international_plan,voice_mail_plan,churn
0,0,0,0
1,0,0,1
2,0,0,1
3,0,0,0
4,0,0,0


In [11]:
clean.columns

Index(['state', 'account_length', 'area_code', 'international_plan',
       'voice_mail_plan', 'number_vmail_messages', 'total_day_minutes',
       'total_day_calls', 'total_day_charge', 'total_eve_minutes',
       'total_eve_calls', 'total_eve_charge', 'total_night_minutes',
       'total_night_calls', 'total_night_charge', 'total_intl_minutes',
       'total_intl_calls', 'total_intl_charge', 'customer_service_calls',
       'churn'],
      dtype='str')

In [12]:
# Work on a copy so the original stays untouched
clean = df.copy()

# Remove duplicate rows (if any)
clean = clean.drop_duplicates()

# Tidy column names: lowercase, underscores instead of spaces
clean.columns = clean.columns.str.strip().str.lower().str.replace(" ", "_")

# Convert Yes/No text to 1/0
clean["international_plan"] = clean["international_plan"].map({"Yes": 1, "No": 0})
clean["voice_mail_plan"] = clean["voice_mail_plan"].map({"Yes": 1, "No": 0})

# Convert True/False to 1/0
clean["churn"] = clean["churn"].astype(int)

# Make sure State is consistent text
clean["state"] = clean["state"].str.strip().str.upper()

clean.head()

,state,account_length,area_code,international_plan,voice_mail_plan,number_vmail_messages,total_day_minutes,total_day_calls,total_day_charge,total_eve_minutes,total_eve_calls,total_eve_charge,total_night_minutes,total_night_calls,total_night_charge,total_intl_minutes,total_intl_calls,total_intl_charge,customer_service_calls,churn
0,LA,117,408,0,0,0,184.5,97,31.37,351.6,80,29.89,215.8,90,9.71,8.7,4,2.35,1,0
1,IN,65,415,0,0,0,129.1,137,21.95,228.5,83,19.42,208.8,111,9.40,12.7,6,3.43,4,1
2,NY,161,415,0,0,0,332.9,67,56.59,317.8,97,27.01,160.6,128,7.23,5.4,9,1.46,4,1
3,SC,111,415,0,0,0,110.4,103,18.77,137.3,102,11.67,189.6,105,8.53,7.7,6,2.08,2,0
4,HI,49,510,0,0,0,119.3,117,20.28,215.1,109,18.28,178.7,90,8.04,11.1,1,3.00,1,0


In [13]:
clean.columns

Index(['state', 'account_length', 'area_code', 'international_plan',
       'voice_mail_plan', 'number_vmail_messages', 'total_day_minutes',
       'total_day_calls', 'total_day_charge', 'total_eve_minutes',
       'total_eve_calls', 'total_eve_charge', 'total_night_minutes',
       'total_night_calls', 'total_night_charge', 'total_intl_minutes',
       'total_intl_calls', 'total_intl_charge', 'customer_service_calls',
       'churn'],
      dtype='str')

In [14]:
clean[["international_plan", "voice_mail_plan", "churn"]].head()

,international_plan,voice_mail_plan,churn
0,0,0,0
1,0,0,1
2,0,0,1
3,0,0,0
4,0,0,0


In [15]:
clean.to_csv("cleaned_churn.csv", index=False)